packages/valory/skills/apy_estimation_abci/models.py

```diff
@@ -167,6 +172,65 @@ def utilized_subgraphs(self) -> ValidatedSubgraphsType:
         return cast(ValidatedSubgraphsMappingType, self._utilized_subgraphs).values()
 
 
+def _raise_incorrect_hacky_initialization() -> None:
+    """
+    Raise a `ValueError` when the pair ids hacky initialization is incorrect.
+
+    Please also take a look at the discussion here:
+    https://github.com/valory-xyz/open-autonomy/pull/1171#discussion_r938924806
+    This method should be removed when https://github.com/valory-xyz/open-aea/issues/270 is closed.
+    """
+    raise ValueError(
+        "Pair ids have been incorrectly configured. Please consider: https://github.com/valory-xyz/open-aea/issues/270."
+    )
+
+
+def _validate_parsed_hacky_pair_ids(pair_ids: PairIdsType) -> None:
+    """
+    Validate the parsed hacky initialized pair ids.
+
+    Please also take a look at the discussion here:
+    https://github.com/valory-xyz/open-autonomy/pull/1171#discussion_r938924806
+    This method should be removed when https://github.com/valory-xyz/open-aea/issues/270 is closed.
+
+    :param pair_ids: the parsed hacky pair ids to validate.
+    """
+    for dex_name, dex_ids in pair_ids.items():
+        if not isinstance(dex_name, str) or not isinstance(dex_ids, list):
+            _raise_incorrect_hacky_initialization()
+        for dex_id in dex_ids:
+            if not isinstance(dex_id, str):
+                _raise_incorrect_hacky_initialization()
+
+
+def _hack_around_dict_override_limitation(pair_ids: HackyPairIdsType) -> PairIdsType:
+    """
+    Temporary hack to overcome the dict override limitation in the agent's config.
+
+    Please also take a look at the discussion here:
+    https://github.com/valory-xyz/open-autonomy/pull/1171#discussion_r938924806
+    This method should be removed when https://github.com/valory-xyz/open-aea/issues/270 is closed.
+
+    :param pair_ids: the hacky pair ids to parse to their normal dict representation.
+    :return: the parsed pair ids.
+    """
+    ids: Dict[str, List[str]] = {}
+
+    try:
+        ids = {
+            # here we make the assumption that the first item contains the key and the second the `list` of `str` values
+            str(dex_config[i]): cast(List[str], dex_config[i + 1])
+            for dex_config in pair_ids
+            for i in range(0, len(dex_config), 2)
+        }
```

Comment:
`ids: Dict[str, List[str]] = dict(zip(dex_config[::2], dex_config[1::2]))`

Reply:
This does not work since we have `pair_ids` which contain many `dex_config`s.

Reply:
?? 

did you try it?

Reply:
So waddaboot `dict(zip(r[::2], r[1::2] for r in x))` then?